### ISSUE TITLE
More flexible S2I binary build data format

### ISSUE THREAD

**Issue by rhuss** [fabric8io/fabric8-maven-plugin#825]
### Description

Currently for an S2I binary build f-m-p sends a `docker-build.tar` as binary input during a binary build. This is the same archive which would be send to a Docker daemon for building an image (and e.g. contains also an (here unused) Dockerfile). 

This works fine with the S2I builder image we have under control (like the FIS images). However other S2I Builder images (like EAP or Wildfly) expect the binary input in a different format. For Java Servlet container (including Java EE server) its the plain artifact (`.war` or probably also `.ear`). 

There should be a way for generators and the build configuration in general to influence this format send to OpenShift. Initially this could be determined by some sort property ("tar", "bin" or so), but could be then even more flexible.

At least for the war generator it should be possible to send the war file as binary to the builder image, because currently the war generation is useless when using s2i.

**Comment by siamaksade** [comment]
This is an important shortcoming since usually, projects have multiple modules and one of which might be a WAR. Lack of support for WAR means people won't have a consistent way to kick the S2I binary build which might make them opt for `oc start-build` instead that works across the board.

**Comment by rhuss** [comment]
Agreed that this quite important and is one of the next thing we are going to tackle.
